packages/tooltips/src/utils/types.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { HTMLAttributes, ReactElement, ReactNode } from 'react';
+import { Modifiers } from 'popper.js';
+
+const SHARED_PLACEMENT = [
+  'auto',
+  'top',
+  'top-start',
+  'top-end',
+  'bottom',
+  'bottom-start',
+  'bottom-end'
+] as const;
+
+export const PLACEMENT = [
+  ...SHARED_PLACEMENT,
+  'end',
+  'end-top',
+  'end-bottom',
+  'start',
+  'start-top',
+  'start-bottom'
+] as const;
+
+export const POPPER_PLACEMENT = [
+  ...SHARED_PLACEMENT,
+  'right',
+  'right-start',
+  'right-end',
+  'left',
+  'left-start',
+  'left-end'
+] as const;
+
+export const SIZE = ['small', 'medium', 'large', 'extra-large'] as const;
+
+export const TYPE = ['light', 'dark'] as const;
+
+export type GardenPlacement = typeof PLACEMENT[number];
+
+export type PopperPlacement = typeof POPPER_PLACEMENT[number];
```

Comment:
See https://melvingeorge.me/blog/convert-array-into-string-literal-union-type-typescript for array-to-type conversion explanation.